browser_history/generic.py

```diff
@@ -167,34 +173,58 @@ def fetch(self, history_paths=None, sort=True, desc=False):
             history_paths = self.history_paths()
         output_object = Outputs()
         with tempfile.TemporaryDirectory() as tmpdirname:
+            length = 0
             for history_path in history_paths:
                 copied_history_path = shutil.copy2(history_path.absolute(), tmpdirname)
                 conn = sqlite3.connect(f"file:{copied_history_path}?mode=ro", uri=True)
                 cursor = conn.cursor()
-                cursor.execute(self.history_SQL)
-                date_histories = [
-                    (
-                        datetime.datetime.strptime(d, "%Y-%m-%d %H:%M:%S").replace(
-                            tzinfo=_local_tz
-                        ),
-                        url,
-                    )
-                    for d, url in cursor.fetchall()
-                ]
-                output_object.entries.extend(date_histories)
+                if type=='history':
+                    cursor.execute(self.history_SQL)
+                    date_histories = [
+                        (
+                            datetime.datetime.strptime(d, "%Y-%m-%d %H:%M:%S").replace(
+                                tzinfo=_local_tz
+                            ),
+                            url,
+                        )
+                        for d, url in cursor.fetchall()
+                    ]
+                    output_object.history_entries.extend(date_histories)
+                    if sort:
+                        output_object.history_entries.sort(reverse=desc)
+                elif type == 'bookmarks':
+                    cursor.execute(self.bookmarks_SQL)
+                    date_bookmarks = [
+                        (
+                            datetime.datetime.strptime(d, "%Y-%m-%d %H:%M:%S").replace(
+                                tzinfo=_local_tz
+                            ),
+                            url,
+                            title,
+                            folder,
+                        )
+                        for d, url,title,folder in cursor.fetchall()
+                    ]
+                    if(len(date_bookmarks)<=length):
+                        continue
+                    length  = len(date_bookmarks)
```

Comment:
This will only add bookmarks whose length was more than the previous added bookmarks. This might skip some bookmarks even though they are valid, since in each iteration of the loop we're looking at a different profile.

Reply:
Currently, there are only redundant copies in both profiles, that's why i did this, I'm not sure for other browsers , haven't checked yet, maybe a bug for the future?